py3status/modules/keyboard_locks.py

```diff
@@ -1,36 +1,39 @@
 # -*- coding: utf-8 -*-
 """
-Monitor CapsLock, NumLock, and ScrLock keys
+Display indicators for CapsLock, NumLock, and ScrLock.
 
 Configuration parameters:
     cache_timeout: refresh interval for this module (default 1)
-    format: display format for this module (default '{caps} {num} {scr}')
-    icon_caps_off: show when Capitals Lock is off (default 'CAPS')
-    icon_caps_on: show when Capitals Lock is on (default 'CAPS')
-    icon_num_off: show when Numeric Lock is off (default 'NUM')
-    icon_num_on: show when Numeric Lock is on (default 'NUM')
-    icon_scr_off: show when Scroll Lock is off (default 'SCR')
-    icon_scr_on: show when Scroll Lock is on (default 'SCR')
+    format: display format for this module
+        (default '[\?if=caps_lock&color=good CAPS|\?color=bad CAPS]
+        [\?if=num_lock&color=good NUM|\?color=bad NUM]
+        [\?if=scroll_lock&color=good SCR|\?color=bad SCR]')
+
+Control placeholders:
+    {caps_lock} a boolean based on xset data
+    {num_lock} a boolean based on xset data
+    {scroll_lock} a boolean based on xset data
 
 Color options:
     color_good: Lock on
     color_bad: Lock off
 
 @author lasers
 
+Examples:
+```
+# hide CAPS, NUM, SCR
+keyboard_locks {
+    format = '\?color=good [\?if=caps_lock CAPS][\?soft  ]'
+    format += '[\?if=num_lock NUM][\?soft  ][\?if=scroll_lock SCR]'
+}
+```
+
 SAMPLE OUTPUT
-[
-    {'color': '#00FF00', 'full_text': 'CAPS '},
-    {'color': '#00FF00', 'full_text': 'NUM '},
-    {'color': '#FF0000', 'full_text': 'SCR'},
-]
+{'color': '#00FF00', 'full_text': 'CAPS NUM SCR'},
```

Comment:
I think it is good to have sample output showing the different colors like we had.